src/mongoose_cluster_id.erl

```diff
@@ -0,0 +1,156 @@
+-module(mongoose_cluster_id).
+
+-include("mongoose.hrl").
+
+-export([
+         start/0,
+         get_cached_cluster_id/0,
+         get_backend_cluster_id/0
+        ]).
+
+% For testing purposes only
+-export([clean_table/0]).
+
+-record(mongoose_cluster_id, {key :: atom(), value :: cluster_id()}).
+-type cluster_id() :: binary().
+-type maybe_cluster_id() :: {ok, cluster_id()} | {error, any()}.
+-type mongoose_backend() :: rdbms
+                          | mnesia.
+
+-spec start() -> maybe_cluster_id().
+start() ->
+    init_mnesia_cache(),
+    run_steps(
+      [fun get_cached_cluster_id/0,
+       fun get_backend_cluster_id/0,
+       fun make_and_set_new_cluster_id/0],
+      {error, no_cluster_id_yet}).
+
+-spec run_steps(ListOfFuns, MaybeCID) -> maybe_cluster_id() when
+      ListOfFuns :: [fun(() -> maybe_cluster_id())],
+      MaybeCID :: maybe_cluster_id().
+run_steps(_, {ok, ID}) when is_binary(ID) ->
+    store_cluster_id(ID);
+run_steps([Fun | NextFuns], {error, _}) ->
+    run_steps(NextFuns, Fun());
+run_steps([], {error, _} = E) ->
+    E.
+
+%% Get cached version
+-spec get_cached_cluster_id() -> maybe_cluster_id().
+get_cached_cluster_id() ->
+    T = fun() -> mnesia:read(mongoose_cluster_id, cluster_id) end,
+    case mnesia:transaction(T) of
+        {atomic, [#mongoose_cluster_id{value = ClusterID}]} ->
+            {ok, ClusterID};
+        {atomic, []} ->
+            {error, cluster_id_not_in_mnesia};
+        {aborted, Reason} ->
+            {error, Reason}
+    end.
+
+%% ====================================================================
+%% Internal getters and setters
+%% ====================================================================
+-spec get_backend_cluster_id() -> maybe_cluster_id().
+get_backend_cluster_id() ->
+    get_cluster_id_from_backend(which_backend_available()).
+
+-spec set_new_cluster_id(cluster_id()) -> maybe_cluster_id().
+set_new_cluster_id(ID) ->
+    set_new_cluster_id(ID, which_backend_available()).
+
+-spec make_and_set_new_cluster_id() -> maybe_cluster_id().
+make_and_set_new_cluster_id() ->
+    NewID = make_cluster_id(),
+    set_new_cluster_id(NewID).
+
+%% ====================================================================
+%% Internal functions
+%% ====================================================================
+init_mnesia_cache() ->
+    mnesia:create_table(mongoose_cluster_id,
+                        [{type, set},
+                         {record_name, mongoose_cluster_id},
+                         {attributes, record_info(fields, mongoose_cluster_id)},
+                         {ram_copies, [node()]}
+                        ]),
+    mnesia:add_table_copy(mongoose_cluster_id, node(), ram_copies).
+
+-spec make_cluster_id() -> cluster_id().
+make_cluster_id() ->
+    uuid:uuid_to_string(uuid:get_v4(), binary_standard).
+
+%% Which backend is enabled
+-spec which_backend_available() -> mongoose_backend().
+which_backend_available() ->
```

Comment:
Considering we have only mnesia or rdbms as backends, I find the whole `try ... catch` ugly here. Could it be refactored to a simple `case` statement?